### PR TITLE
fix(mdraid): try to assemble the missing raid device

### PR DESCRIPTION
**Pull request by ldzhong**
If some raid devices with specified UUIDs fail to be assembled in initrd, we will try to assemble them again in this script to avoid system falling into emergency mode. This patch is created because we can offen see mdadm command failure during boot because of timing issue introduced between mdadm and udevd.

This pull request changes...

## Checklist
- [ ] I have tested it locally
- [ ] I have reviewed and updated any documentation if relevant
- [ ] I am providing new code and test(s) for it

Fixes #

### CODE REVIEW

File: modules.d/90mdraid/mdraid_start.sh [hunk 1]
```diff
@@ -65,6 +70,14 @@ _md_force_run() {
             _md_start "${_md}"
         done
     fi
+
+    # try to assemble all the left md devices that should be assembled
```
Comment (aafeijoo-suse) [inline modules.d/90mdraid/mdraid_start.sh:74]: Remove white space after the word assembled:

https://github.com/dracut-ng/dracut-ng/actions/runs/9638464635/job/26579189718?pr=410

File: modules.d/90mdraid/mdraid_start.sh [hunk 2]
```diff
@@ -65,6 +70,14 @@ _md_force_run() {
             _md_start "${_md}"
         done
     fi
+
+    # try to assemble all the left md devices that should be assembled 
+    # probably failed because of timing issue in the first beginning
+    if [ -n "$_LEFT_UUID" ]; then
+        for _uuid in ${_LEFT_UUID}; do
```
Comment (aafeijoo-suse) [inline modules.d/90mdraid/mdraid_start.sh:77]: Add `local _uuid` to the beginning of the function, where the other local variables are defined.

**Review by aafeijoo-suse — APPROVED** [review]
Also you need to fix the commit message, to something like: `fix(mdraid): try to assemble the missing raid device`

https://github.com/dracut-ng/dracut-ng/actions/runs/9638472397/job/26579207548?pr=410

Those are minor nitpicks to comply with the coding style, other than that your patch is good. Thanks Lidong!

**Comment by ldzhong** [comment]
Thanks for your review, Antonio. I have made the changes as you suggested.

**Review by LaszloGombos — APPROVED** [review]

**Pull request merged**